front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts

```diff
@@ -235,22 +236,40 @@ export class CreateF3XStep1Component implements OnInit, OnDestroy {
       summary.form_type = F3xFormTypes.F3XT;
     }
 
-    this.f3xSummaryService.create(summary, this.formProperties).subscribe((report: F3xSummary) => {
-      if (jump === 'continue') {
-        // Save report to Cash On Hand in the store if necessary by pulling the reports table data.
-        this.reportService.getTableData().subscribe();
+    //Observables are *defined* here ahead of their execution
+    const create = this.f3xSummaryService.create(summary, this.formProperties);
+    // Save report to Cash On Hand in the store if necessary by pulling the reports table data.
+    const tableData = this.reportService.getTableData();
+    const cashOnHand = this.store.select(selectCashOnHand);
 
-        this.router.navigateByUrl(`/reports/f3x/create/step2/${report.id}`);
-      } else {
-        this.router.navigateByUrl('/reports');
-        this.messageService.add({
-          severity: 'success',
-          summary: 'Successful',
-          detail: 'Contact Updated',
-          life: 3000,
-        });
-      }
-    });
+    //Create the report, update cashOnHand based on all reports, and then retrieve cashOnHand in that order
+    create
+      .pipe(
+        switchMap((report) => {
+          return zip(of(report), tableData);
+        }),
+        switchMap(([report, _]) => {
+          return zip(of(report), cashOnHand);
+        }),
```

Comment:
This seems like a lot of work to subscribe to 3 independent observables. Did you consider forkJoin()?  ... and maybe tap() the tableData$ so not to have the extra _ array element?

Something like:
forkJoin([create$, cashOnHand$].pipe(tap(tableData$.subscribe()).subscribe([report, coh]) => {}

Reply:
the issue with using tap here is that we wouldn't get the assured sequence of operation since tap is for side effects and returns the original observable.  We need it to return a new observable dependant on getTableData$ completing

Reply:
:+1: